android/src/main/kotlin/com/mapbox/maps/mapbox_maps/ProxyBinaryMessenger.kt

```diff
@@ -0,0 +1,25 @@
+package com.mapbox.maps.mapbox_maps
+
+import io.flutter.plugin.common.BinaryMessenger
+import java.nio.ByteBuffer
+
+// BinaryMessenger that proxies all calls to another BinaryMessenger,
+// but with a suffix appended to the channel name.
+// Different map instances use different suffixes to avoid channel name conflicts.
+class ProxyBinaryMessenger(
+  private val messenger: BinaryMessenger,
+  private val nameSuffix: String,
+) : BinaryMessenger {
+
+  override fun send(channel: String, message: ByteBuffer?) {
+    messenger.send("$channel$nameSuffix", message)
```

Comment:
```suggestion
    messenger.send(channelName, message)
```

Reply:
Cant extract it here - channel is a function call parameter and changes.

Reply:
ah, yes
you can add some like that:
```
private fun String.toChannelName() = "$this$nameSuffix"
```
inside class. just not to forgot to change each place when you will make some refactoring in the future

Reply:
Done